interfaces/policy/basedeclaration.go

```diff
@@ -658,9 +669,45 @@ slots:
         - core
 `
 
+func ComposeBaseDeclaration(ifaces []interfaces.Interface) ([]byte, error) {
+	var buf bytes.Buffer
+	// Trim newlines at the end of the string. All the elements may have
+	// spurious trailing newlines. All elements start with a leading newline.
+	// We don't want any blanks as that would no longer parse.
+	tr := func(s string) string { return strings.TrimRight(s, "\n") }
```

Comment:
Would like a better name for `tr` here as it doesn't say much if you look at something like `tr(baseDeclarationHeader)`. Something like `trimTrailingNewLine` or so would make it obvious fo the reader.